src/transaction.ts

```diff
@@ -161,7 +197,53 @@ class Transaction extends DatastoreRequest {
         : () => {};
     const gaxOptions =
       typeof gaxOptionsOrCallback === 'object' ? gaxOptionsOrCallback : {};
-    this.#runCommit(gaxOptions, callback);
+    const resolver: Resolver<google.datastore.v1.ICommitResponse> = resolve => {
+      this.#runCommit(
+        gaxOptions,
+        (err?: Error | null, resp?: google.datastore.v1.ICommitResponse) => {
+          resolve({err, resp});
+        }
+      );
+    };
+    this.#withBeginTransaction(gaxOptions, resolver).then(
+      (response: UserCallbackData<google.datastore.v1.ICommitResponse>) => {
+        callback(response.err, response.resp);
+      }
+    );
+  }
+
+  /**
+   * If the transaction has not begun yet then this function ensures the transaction
+   * has started before running the resolver provided. The resolver is a function with one
+   * argument. This argument is a function that is used to pass errors and
+   * response data back to the caller of the withBeginTransaction function.
+   *
+   * @param {CallOptions | undefined} [gaxOptions]
+   * @param {Resolver<T>} [resolver]
+   * @returns {Promise<UserCallbackData<T>>} Returns a promise that will run
+   * this code and resolve to an error or resolve with the data from the resolver.
+   * @private
+   */
+  async #withBeginTransaction<T>(
+    gaxOptions: CallOptions | undefined,
+    resolver: Resolver<T>
+  ): Promise<UserCallbackData<T>> {
+    if (this.#state === TransactionState.NOT_STARTED) {
+      const release = await this.#mutex.acquire();
+      try {
```

Comment:
Nitpick: is it possible to make this a "try {} catch {} finally {}" block instead of using nested try blocks?

Reply:
Yup. Done.